wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet/Create.hs

```diff
@@ -84,33 +95,34 @@ createHdRoot rootId name hasPass assurance created =
         , _hdRootCreatedAt   = created
         }
 
--- | Create a new account in the specified wallet
+-- | Create a new account in the specified wallet, unless it already exists
 --
 -- It is the responsibility of the caller to check the wallet's spending
 -- password.
 --
 -- TODO: If any key derivation is happening when creating accounts, should we
 -- store a public key or an address or something?
-createHdAccount :: HdRootId
-                -> AccountName
+createHdAccount :: HdAccountId
                 -> Checkpoint
-                -> Update' HdWallets CreateHdAccountError HdAccountId
-createHdAccount rootId name checkpoint = do
+                -> Update' HdWallets CreateHdAccountError ()
+createHdAccount accountId checkpoint = do
     -- Check that the root ID exists
-    zoomHdRootId CreateHdAccountUnknown rootId $
+    zoomHdRootId CreateHdAccountUnknownRoot rootId $
       return ()
-    -- Create the new account
+
     zoom hdWalletsAccounts $ do
-      numAccounts <- gets $ IxSet.size . IxSet.getEQ rootId
-      let accIx = HdAccountIx (fromIntegral numAccounts)
-          accId = HdAccountId rootId accIx
-      at accId .= Just (hdAccount accId)
-      return accId
+        exists <- gets $ IxSet.member accountId
+        unless exists $ -- TODO add callback function for accountExists (to enable caller to throw Err)
```

Comment:
Let's not leave this TODO here, but why do we need a callback? Why not throw an error, like the other create functions do?